similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -151,16 +155,26 @@ public int compare(PatientSimilarityView o1, PatientSimilarityView o2)
      * @param referencePatient the reference patient
      * @return a query populated with terms from the patient phenotype
      */
+    @SuppressWarnings("unchecked")
     private SolrQuery generateQuery(Patient referencePatient, boolean prototypes)
     {
+        SolrQuery queryt = new SolrQuery();
+        StringBuilder qr = new StringBuilder();
+        qr.append("(document:" + ClientUtils.escapeQueryChars(referencePatient.getDocument().toString()) + " )");
```

Comment:
This crashes with an NPE when the referencePatient does not have a document (e.g., if it's a RemoteMatchingPatient). Should compute the features and ancestors on the fly.